File: Arrays_n_Strings/find_target.js
/* Works with a sorted array
-----------------------------
 @param {number []} array arr
 @param {number} target
 @return {boolean}
 */

function findTargetedSum(arr, target) {
    let left = 0;
    let right = arr.length - 1;

    while (left < right) {
        let currentSum = arr[left] + arr[right];
        
        if (currentSum == target) {
            return true;
        }
        if (currentSum > target) {
            right--;
        } else {
            left++;
        }
    
    }
    return false;
}

console.log(findTargetedSum([35, 42, 46, 73, 99], 88))

File: Arrays_n_Strings/find_target.py
# This works for a sorted array
def findTarget(arr, target) -> bool:
    left = 0
    right = len(arr) - 1

    while left < right:
        currentSum = arr[left] + arr[right]
        if currentSum == target:
            return True
        if currentSum > target:
            right -= 1
        else:
            left += 1
    return False

print(findTarget([45, 67, 78, 80], 125))